### PR TITLE
DOMA-3251 Prohibition on the use of weak passwords

### PR DESCRIPTION
**Pull request by pahaz**
image
image
image

### CODE REVIEW

**Review by toplenboren — APPROVED** [review]

File: apps/condo/domains/user/schema/ForgotPasswordService.test.js [hunk 1]
```diff
@@ -237,6 +237,26 @@ describe('ForgotPasswordAction Service', () => {
             }])
         })
         
+        it('cannot change password to weak passord', async () => {
```
Comment (akarjakin) [inline apps/condo/domains/user/schema/ForgotPasswordService.test.js:240]: passord -> password

**Review by nomerdvadcatpyat — APPROVED** [review]

**Review by akarjakin — APPROVED** [review]

File: apps/condo/domains/common/utils/codegeneration/generate.server.utils.js [hunk 2]
```diff
@@ -1,21 +1,67 @@
...
 const IS_DEBUG = conf.NODE_ENV === 'development' || conf.NODE_ENV === 'test'
 
 const isNotUndefined = (x) => typeof x !== 'undefined'
+const ALLOWED_OPTIONS = ['errorMapping', 'doesNotExistError', 'multipleObjectsError']
 
-function _throwIfError (context, errors, data, errorMessage) {
+function _getAllErrorMessages (errors) {
+    const messages = []
+    for (const x of errors) {
+        const m = get(x, 'message', '')
```
Comment (sitozzz) [inline apps/condo/domains/common/utils/codegeneration/generate.server.utils.js:15]: There is no need to use default value here
```suggestion
        const m = get(x, 'message')
```

**Comment by toplenboren** [comment]
I'd rather not say that this password is in the list of stolen passwords. Why user needs to know that? 

Lets simply say, `this password is weak or too obvious please choose another password`

File: apps/condo/domains/common/utils/codegeneration/generate.server.utils.js [hunk 3]
```diff
@@ -1,21 +1,67 @@
-const { pickBy, get, isEmpty } = require('lodash')
+const { pickBy, get, isEmpty, isObject } = require('lodash')
 
 const conf = require('@core/config')
 const { getById } = require('@core/keystone/schema')
+const { GQLError } = require('@core/keystone/errors')
 
 const IS_DEBUG = conf.NODE_ENV === 'development' || conf.NODE_ENV === 'test'
 
 const isNotUndefined = (x) => typeof x !== 'undefined'
```
[the remark this answers is not in the document]
Comment (pahaz) [reply apps/condo/domains/common/utils/codegeneration/generate.server.utils.js:9]: yes ... but could you please check the usecase for that func and you will see why it so

**Pull request merged**